src/fdb5/database/EntryVisitMechanism.h

```diff
@@ -52,7 +51,7 @@ class EntryVisitor : public eckit::NonCopyable {
 
 private: // methods
 
-    virtual void visitDatum(const Field& field, const Key& key) = 0;
+    virtual void visitDatum(const Field& field, const TypedKey& datumKey) = 0;
```

Comment:
critical decision to choose `TypeKey` here..
if all visitDatum implementations need `Key` (hence each calls canocinal), then leave it as `Key` and let parent do it ..